### PR TITLE
task 9363: fetch and display data api

### PR DESCRIPTION
**Pull request by vanpd-0859**
### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/9363
### Goals :soccer:

Fetch data from API and display on UI:
- Asset information
- List asset transactions

### Implementation Details :construction:

https://gfycat.com/OnlyPrestigiousChamois

### CODE REVIEW

**Comment by vanpd-0859** [comment]
ready

File: CryptoWallet/Data/Source/Remote/API/Request/TransactionRequest.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+final class TransactionRequest: BaseRequest {
+    required init(walletAddress: String, page: Int, contractAddress: String) {
+        var body: [String: Any] = [
+            "address": walletAddress,
+            "page": page
+        ]
+        if !contractAddress.isEmpty {
+            body["contract"] = contractAddress
+        }
+        print(body)
```
Comment (tienpx3010) [inline CryptoWallet/Data/Source/Remote/API/Request/TransactionRequest.swift]: Bỏ

File: CryptoWallet/Scenes/Home/Wallet/Asset/AssetViewController.swift [hunk 2]
```diff
@@ -82,13 +85,126 @@ final class AssetViewController: UIViewController {
...
+            return
+        }
+        if assetInfo.type == .coin {
+            transactionRepository.getTransactionList(walletAddress: wallet.walletAddress, page: page) { (result) in
+                switch result {
+                case .success(let transactionResponse):
+                    guard let transactionList = transactionResponse?.transactions else {
+                        return
+                    }
+                    self.transactionList = transactionList
```
Comment (tienpx3010) [inline CryptoWallet/Scenes/Home/Wallet/Asset/AssetViewController.swift:193]: weak self

**Comment by vanpd-0859** [comment]
ready

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**